modules/light-clients/07-tendermint/types/update.go

```diff
@@ -60,6 +60,34 @@ func (cs ClientState) CheckHeaderAndUpdateState(
 		return nil, nil, err
 	}
 
+	// Check the earliest consensus state to see if it is expired, if so then set the prune height
+	// so that we can delete consensus state and all associated metadata.
```

Comment:
can you add a test for this logic? I guess it can wait until the testing refactor part 2 is merged